src/utils/iap.ts

```diff
@@ -0,0 +1,11 @@
+import { GoogleAuth } from 'google-auth-library';
+
+import { IAP_TARGET_AUDIENCE } from '@/config';
+
+const auth = new GoogleAuth();
+
+export async function getIDToken(): Promise<string> {
+  const client = await auth.getIdTokenClient(IAP_TARGET_AUDIENCE);
```

Comment:
I think you can use the client here to make requests instead of using fetch actually, so we can use one less library 🤔

Reply:
The problem I ran into is that when I used the client, it would inject the id token as an authorization instead of proxy-authorization. I can look into this some more though